File: android/src/main/java/org/ligi/gobandroid_hd/ui/gnugo/GnuGoGame.java
package org.ligi.gobandroid_hd.ui.gnugo;

import org.ligi.gobandroid_hd.App;
import org.ligi.gobandroid_hd.R;
import org.ligi.gobandroid_hd.logic.GoGame;
import org.ligi.gobandroid_hd.logic.GoGameMetadata;
import org.ligi.gobandroid_hd.ui.GoPrefs;

public class GnuGoGame {

    public final boolean playingBlack;
    public final boolean playingWhite;
    public final byte level;
    private final GoGame game;

    public boolean aiIsThinking = false;

    GnuGoGame(final boolean playingBlack, final boolean playingWhite, final byte level, final GoGame game) {
        App.component().inject(this);
        this.playingBlack = playingBlack;
        this.playingWhite = playingWhite;
        this.level = level;
        this.game = game;
    }

    public boolean gnugoNowWhite() {
        return !game.isBlackToMove() && playingWhite;
    }

    public boolean gnugoNowBlack() {
        return game.isBlackToMove() && playingBlack;
    }

    public void setMetaDataForGame(App app) {
        final GoGameMetadata metaData = game.getMetaData();
        if (playingBlack) {
            metaData.setBlackName(app.getString(R.string.gnugo));
            metaData.setBlackRank("");
        } else {
            metaData.setBlackName(GoPrefs.INSTANCE.getUsername());
            metaData.setBlackRank(GoPrefs.INSTANCE.getRank());
        }

        if (playingWhite) {
            metaData.setWhiteName(app.getString(R.string.gnugo));
            metaData.setWhiteRank("");
        } else {
            metaData.setWhiteName(GoPrefs.INSTANCE.getUsername());
            metaData.setWhiteRank(GoPrefs.INSTANCE.getRank());
        }

    }
}

File: android/src/main/java/org/ligi/gobandroid_hd/ui/gnugo/GnuGoGame.kt
package org.ligi.gobandroid_hd.ui.gnugo

import org.ligi.gobandroid_hd.App
import org.ligi.gobandroid_hd.R
import org.ligi.gobandroid_hd.logic.GoGame
import org.ligi.gobandroid_hd.ui.GoPrefs

class GnuGoGame internal constructor(val playingBlack: Boolean, val playingWhite: Boolean, val level: Byte, private val game: GoGame) {

    var aiIsThinking = false

    fun gnugoNowWhite(): Boolean {
        return !game.isBlackToMove && playingWhite
    }

    fun gnugoNowBlack(): Boolean {
        return game.isBlackToMove && playingBlack
    }

    fun setMetaDataForGame(app: App) {
        val metaData = game.metaData
        if (playingBlack) {
            metaData.blackName = app.getString(R.string.gnugo)
            metaData.blackRank = ""
        } else {
            metaData.blackName = GoPrefs.username
            metaData.blackRank = GoPrefs.rank
        }

        if (playingWhite) {
            metaData.whiteName = app.getString(R.string.gnugo)
            metaData.whiteRank = ""
        } else {
            metaData.whiteName = GoPrefs.username
            metaData.whiteRank = GoPrefs.rank
        }

    }
}
